### PR TITLE
Rename some structures and fields in VM subsystem

### PR DESCRIPTION
**Pull request by franciscozdo**
I think it is time to clean names in our VM subsystem. It will be also easier to compare our system with BSD because structures will have similar names.

Changes:
*  renamed fields of `vm_object`
* `vm_segment` -> `vm_map_entry`

### CODE REVIEW

File: include/sys/vm.h [hunk 1]
```diff
@@ -51,7 +51,7 @@ typedef uintptr_t vm_offset_t;
 /* Field marking and corresponding locks:
  * (@) pv_list_lock (in pmap.c)
  * (P) physmem_lock (in vm_physmem.c)
- * (O) vm_object::mtx */
+ * (O) vm_object::uo_lock */
```
Comment (wiklam) [inline include/sys/vm.h:54]:
```suggestion
 * (O) vm_object::vo_lock */
```

**Review by wiklam — APPROVED** [review]
The rest seems ok

File: sys/tests/vm_map.c [hunk 2]
```diff
@@ -102,15 +102,15 @@ static int findspace_demo(void) {
...
   vaddr_t t;
   int n;
 
-  seg = vm_segment_alloc(NULL, addr1, addr2, VM_PROT_NONE, VM_SEG_PRIVATE);
+  seg = vm_map_entry_alloc(NULL, addr1, addr2, VM_PROT_NONE, VM_ENT_PRIVATE);
   n = vm_map_insert(umap, seg, VM_FIXED);
   assert(n == 0);
 
-  seg = vm_segment_alloc(NULL, addr3, addr4, VM_PROT_NONE, VM_SEG_PRIVATE);
+  seg = vm_map_entry_alloc(NULL, addr3, addr4, VM_PROT_NONE, VM_ENT_PRIVATE);
```
Comment (pj1031999) [inline sys/tests/vm_map.c:113]: seg -> ent

Comment (franciscozdo) [reply sys/tests/vm_map.c]: done

**Review by cahirwpz — APPROVED** [review]

**Pull request merged**